driver/ni/ni.cpp

```diff
@@ -8,82 +8,504 @@
+    {"DAQmx_CO_AutoIncrCnt", "auto_incr_cnt"},
+    {"DAQmx_CO_CtrTimebaseMasterTimebaseDiv", "ctr_timebase_master_timebase_div"},
+    {"DAQmx_CO_PulseDone", "pulse_done"},
+    {"DAQmx_CO_EnableInitialDelayOnRetrigger", "enable_initial_delay_on_retrigger"},
+    {"DAQmx_CO_ConstrainedGenMode", "constrained_gen_mode"},
+    {"DAQmx_CO_UseOnlyOnBrdMem", "use_only_on_brd_mem"},
+    {"DAQmx_CO_DataXferMech", "data_xfer_mech"},
+    {"DAQmx_CO_DataXferReqCond", "data_xfer_req_cond"},
+    {"DAQmx_CO_UsbXferReqSize", "usb_xfer_req_size"},
+    {"DAQmx_CO_UsbXferReqCount", "usb_xfer_req_count"},
+    {"DAQmx_CO_MemMapEnable", "mem_map_enable"},
+    {"DAQmx_CO_Prescaler", "prescaler"},
+    {"DAQmx_CO_RdyForNewVal", "rdy_for_new_val"},
+    {"DAQmx_ChanType", "chan_type"},
+    {"DAQmx_PhysicalChanName", "physical_chan_name"},
+    {"DAQmx_ChanDescr", "chan_descr"},
+    {"DAQmx_ChanIsGlobal", "chan_is_global"},
+    {"DAQmx_Chan_SyncUnlockBehavior", "chan_sync_unlock_behavior"},
+    {"DAQmx_SampClk_Rate", "sample_rate"}
 };
 
+
 ///////////////////////////////////////////////////////////////////////////////////
 //                                    NiSource                                   //
 ///////////////////////////////////////////////////////////////////////////////////
-
-
-void ni::Source::getIndexKeys(){
+void ni::Source::get_index_keys() {
     std::set<std::uint32_t> index_keys;
-    //iterate through channels in reader config
-    for (auto &channel : this->reader_config.channels){
-        LOG(INFO) << "constructing index channel configs";
-        auto [channel_info, err] = this->ctx->client->channels.retrieve(channel.channel_key);
-        // TODO handle error with breaker
-        if (err != freighter::NIL){
-            // Log error
-            LOG(ERROR) << "[NI Reader] failed to retrieve channel " << channel.channel_key;
-            this->ok_state = false;
+    for (auto &channel: this->reader_config.channels) {
+        auto [channel_info, err] = this->ctx->client->channels.retrieve(
+            channel.channel_key);
+        if (err) {
+            this->log_error(
+                "failed to retrieve channel " + std::to_string(channel.channel_key));
             return;
-        } else{
-            index_keys.insert(channel_info.index);
-        }
+        } 
+        index_keys.insert(channel_info.index);
     }
-
-    // now iterate through the set and add all the index channels as configs
-    for (auto it = index_keys.begin(); it != index_keys.end(); ++it){
+    for (auto it = index_keys.begin(); it != index_keys.end(); ++it) {
         auto index_key = *it;
-        LOG(INFO) << "constructing index channel configs";
         auto [channel_info, err] = this->ctx->client->channels.retrieve(index_key);
-        if (err != freighter::NIL){
-            LOG(ERROR) << "[NI Reader] failed to retrieve channel " << index_key;
-            this->ok_state = false;
+        if (err != freighter::NIL) {
```

Comment:
The boolean operator on `freighter:Error` already compares it against nil. Just use `if(err)` instead